1.0版使用yml配置时启动异常

- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
引入seata-all或者seata-spring-boot-starter时,按照script文件夹内的yml配置时启动异常

### Ⅱ. Describe what happened

```
org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'globalTransactionScanner' defined in class path resource [io/seata/spring/boot/autoconfigure/SeataAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.seata.spring.annotation.GlobalTransactionScanner]: Factory method 'globalTransactionScanner' threw exception; nested exception is java.lang.ExceptionInInitializerError
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:627) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:456) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1321) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1160) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:204) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.registerBeanPostProcessors(PostProcessorRegistrationDelegate.java:228) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.registerBeanPostProcessors(AbstractApplicationContext.java:721) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:534) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141) ~[spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:744) [spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:391) [spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:312) [spring-boot-2.1.8.RELEASE.jar:2.1.8.RELEASE]
	at org.test.ProviderApplication.main(ProviderApplication.java:22) [classes/:?]
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [io.seata.spring.annotation.GlobalTransactionScanner]: Factory method 'globalTransactionScanner' threw exception; nested exception is java.lang.ExceptionInInitializerError
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	... 17 more
Caused by: java.lang.ExceptionInInitializerError
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:90) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:133) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:122) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration.globalTransactionScanner(SeataAutoConfiguration.java:57) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.CGLIB$globalTransactionScanner$1(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a$$FastClassBySpringCGLIB$$d8c42c73.invoke(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.globalTransactionScanner(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	... 17 more
Caused by: java.lang.NullPointerException
	at io.seata.config.FileConfiguration.<init>(FileConfiguration.java:96) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.config.ConfigurationFactory.<clinit>(ConfigurationFactory.java:58) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:90) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:133) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.annotation.GlobalTransactionScanner.<init>(GlobalTransactionScanner.java:122) ~[seata-all-1.0.0-SNAPSHOT.jar:1.0.0-SNAPSHOT]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration.globalTransactionScanner(SeataAutoConfiguration.java:57) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.CGLIB$globalTransactionScanner$1(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a$$FastClassBySpringCGLIB$$d8c42c73.invoke(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244) ~[spring-core-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363) ~[spring-context-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at io.seata.spring.boot.autoconfigure.SeataAutoConfiguration$$EnhancerBySpringCGLIB$$6d0d178a.globalTransactionScanner(<generated>) ~[seata-spring-boot-starter-1.0.0-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_201]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_201]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_201]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622) ~[spring-beans-5.1.9.RELEASE.jar:5.1.9.RELEASE]
	... 17 more

```
### Ⅳ. How to reproduce it (as minimally and precisely as possible)

引入按yml配置即可重现

- JDK version :1.8
- OS :windows
- Others:

--- a/config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java
+++ b/config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java
@@ -145,28 +145,28 @@ public class FileConfiguration extends AbstractConfiguration {
         }
         ConfigFuture configFuture = new ConfigFuture(dataId, defaultValue, ConfigOperation.GET, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (String) configFuture.get();
+        return (String)configFuture.get();
     }
 
     @Override
     public boolean putConfig(String dataId, String content, long timeoutMills) {
         ConfigFuture configFuture = new ConfigFuture(dataId, content, ConfigOperation.PUT, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (Boolean) configFuture.get();
+        return (Boolean)configFuture.get();
     }
 
     @Override
     public boolean putConfigIfAbsent(String dataId, String content, long timeoutMills) {
         ConfigFuture configFuture = new ConfigFuture(dataId, content, ConfigOperation.PUTIFABSENT, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (Boolean) configFuture.get();
+        return (Boolean)configFuture.get();
     }
 
     @Override
     public boolean removeConfig(String dataId, long timeoutMills) {
         ConfigFuture configFuture = new ConfigFuture(dataId, null, ConfigOperation.REMOVE, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (Boolean) configFuture.get();
+        return (Boolean)configFuture.get();
     }
 
     @Override

--- a/config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java
+++ b/config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java
@@ -16,6 +16,7 @@
 package io.seata.config;
 
 import java.io.File;
+import java.net.URL;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
@@ -71,6 +72,16 @@ public class FileConfiguration extends AbstractConfiguration {
     private final boolean allowDynamicRefresh;
 
     /**
+     * Note that:this constructor is only used to create proxy with CGLIB
+     * see io.seata.spring.boot.autoconfigure.provider.SpringBootConfigurationProvider#provide
+     */
+    public FileConfiguration() {
+        this.name = null;
+        this.targetFilePath = null;
+        this.allowDynamicRefresh = false;
+    }
+
+    /**
      * Instantiates a new File configuration.
      *
      * @param name the name
@@ -92,13 +103,22 @@ public class FileConfiguration extends AbstractConfiguration {
             targetFilePath = name.substring(SYS_FILE_RESOURCE_PREFIX.length());
             Config appConfig = ConfigFactory.parseFileAnySyntax(new File(targetFilePath));
             fileConfig = ConfigFactory.load(appConfig);
+            this.allowDynamicRefresh = allowDynamicRefresh;
         } else {
-            targetFilePath = this.getClass().getClassLoader().getResource(name).getPath();
-            fileConfig = ConfigFactory.load(name);
+            //conf files may not exist when using seata-spring-boot-starter
+            URL resource = this.getClass().getClassLoader().getResource(name);
+            if (null != resource) {
+                targetFilePath = resource.getPath();
+                fileConfig = ConfigFactory.load(name);
+                targetFileLastModified = new File(targetFilePath).lastModified();
+                this.allowDynamicRefresh = allowDynamicRefresh;
+            } else {
+                targetFilePath = null;
+                fileConfig = ConfigFactory.load();
+                this.allowDynamicRefresh = false;
+            }
         }
         this.name = name;
-        this.allowDynamicRefresh = allowDynamicRefresh;
-        targetFileLastModified = new File(targetFilePath).lastModified();
         configOperateExecutor = new ThreadPoolExecutor(CORE_CONFIG_OPERATE_THREAD, MAX_CONFIG_OPERATE_THREAD,
             Integer.MAX_VALUE, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(),
             new NamedThreadFactory("configOperate", MAX_CONFIG_OPERATE_THREAD));

--- a/config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java
+++ b/config/seata-config-core/src/main/java/io/seata/config/FileConfiguration.java
@@ -87,8 +87,9 @@ public class FileConfiguration extends AbstractConfiguration {
      * @param name the name
      */
     public FileConfiguration(String name) {
-        this(name,true);
+        this(name, true);
     }
+
     /**
      * Instantiates a new File configuration.
      *
@@ -100,24 +101,36 @@ public class FileConfiguration extends AbstractConfiguration {
         if (null == name) {
             throw new IllegalArgumentException("name can't be null");
         } else if (name.startsWith(SYS_FILE_RESOURCE_PREFIX)) {
-            targetFilePath = name.substring(SYS_FILE_RESOURCE_PREFIX.length());
-            Config appConfig = ConfigFactory.parseFileAnySyntax(new File(targetFilePath));
-            fileConfig = ConfigFactory.load(appConfig);
-            this.allowDynamicRefresh = allowDynamicRefresh;
+            File targetFile = new File(name.substring(SYS_FILE_RESOURCE_PREFIX.length()));
+            if (targetFile.exists()) {
+                targetFilePath = targetFile.getPath();
+                Config appConfig = ConfigFactory.parseFileAnySyntax(targetFile);
+                fileConfig = ConfigFactory.load(appConfig);
+            } else {
+                targetFilePath = null;
+            }
         } else {
-            //conf files may not exist when using seata-spring-boot-starter
             URL resource = this.getClass().getClassLoader().getResource(name);
             if (null != resource) {
                 targetFilePath = resource.getPath();
                 fileConfig = ConfigFactory.load(name);
-                targetFileLastModified = new File(targetFilePath).lastModified();
-                this.allowDynamicRefresh = allowDynamicRefresh;
+
             } else {
                 targetFilePath = null;
-                fileConfig = ConfigFactory.load();
-                this.allowDynamicRefresh = false;
             }
         }
+        /**
+         * For seata-server side the conf file should always exists.
+         * For application(or client) side,conf file may not exists when using seata-spring-boot-starter
+         */
+        if (null == targetFilePath) {
+            fileConfig = ConfigFactory.load();
+            this.allowDynamicRefresh = false;
+        } else {
+            targetFileLastModified = new File(targetFilePath).lastModified();
+            this.allowDynamicRefresh = allowDynamicRefresh;
+        }
+
         this.name = name;
         configOperateExecutor = new ThreadPoolExecutor(CORE_CONFIG_OPERATE_THREAD, MAX_CONFIG_OPERATE_THREAD,
             Integer.MAX_VALUE, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<>(),
@@ -132,28 +145,28 @@ public class FileConfiguration extends AbstractConfiguration {
         }
         ConfigFuture configFuture = new ConfigFuture(dataId, defaultValue, ConfigOperation.GET, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (String)configFuture.get();
+        return (String) configFuture.get();
     }
 
     @Override
     public boolean putConfig(String dataId, String content, long timeoutMills) {
         ConfigFuture configFuture = new ConfigFuture(dataId, content, ConfigOperation.PUT, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (Boolean)configFuture.get();
+        return (Boolean) configFuture.get();
     }
 
     @Override
     public boolean putConfigIfAbsent(String dataId, String content, long timeoutMills) {
         ConfigFuture configFuture = new ConfigFuture(dataId, content, ConfigOperation.PUTIFABSENT, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (Boolean)configFuture.get();
+        return (Boolean) configFuture.get();
     }
 
     @Override
     public boolean removeConfig(String dataId, long timeoutMills) {
         ConfigFuture configFuture = new ConfigFuture(dataId, null, ConfigOperation.REMOVE, timeoutMills);
         configOperateExecutor.submit(new ConfigOperateRunnable(configFuture));
-        return (Boolean)configFuture.get();
+        return (Boolean) configFuture.get();
     }
 
     @Override

